### DISCUSSION TITLE
How are transparent images handled in Arduino_GFX?

### DISCUSSION THREAD

**Discussion by nielsnl68** [moononournation/Arduino_GFX#253] [Q&A]
Hello again,
Do you have anything included that should add transparently in images? And how do you have done that?

**Comment by h5n1xp** [comment]
As far as I can see no transparency method is supported by this library yet.

Alpha blending is quite computationally expensive (especially in 16bit gfx) and could only work if you are using a canvas where the framebuffer can be read back.

I'm currently thinking about how to subclass the canvas type to implement sprites which will need some kind of transparency, I'm thinking that colour-keying would make the most sense here (as stencil masks and alphas will require too much CPU/RAM bandwidth).

**Comment by moononournation** [comment]
The PNG image viewer example is transparent background, is it something you want?
